trailmap/acquisition_paths.py

```diff
@@ -0,0 +1,42 @@
+import networkx as nx
+from pprint import pprint
+
+
+def conduct_apa(facility_dict_in, facility_dict_out):
+    ''' '''
+    G = build_graph(facility_dict_in, facility_dict_out)
+
+    sources = list(node for node, in_deg in G.in_degree() if in_deg == 0)
+    sinks = list(node for node, out_deg in G.out_degree() if out_deg == 0)
+
+    print("\nSimple paths")
+    pathways = find_simple_paths(G, sources, sinks)
+    print(pathways)
+
+    return G, pathways
+
+
+def build_graph(facility_dict_in, facility_dict_out):
+    '''Builds NetworkX graph from Cyclus input file
+    '''
+    G = nx.MultiDiGraph()
+    G.add_nodes_from(facility_dict_in.keys())
+    for receiver, incommods in facility_dict_in.items():
+        # find all facilities with that commod as an outcommod
+        for sender, outcommods in facility_dict_out.items():
+            connection = list(set(incommods).intersection(outcommods))
+            for commod in connection:
+                G.add_edge(sender, receiver, commodity=commod)
+    return G
+
+
+def find_simple_paths(G, sources, sinks):
+    ''' finds all simple paths between a given list of sources and targets
+    '''
+    pathways = set()
+    for source, sink in [(x, y) for x in sources for y in sinks]:
+        paths = list(nx.all_simple_paths(G, source=source, target=sink))
+        for path in paths:
+            pathways.add(tuple(path))
```

Comment:
```suggestion
    for source, sink in [(x, y) for x in sources for y in sinks]:
        pathways += tupple(nx.all_simple_paths(G, source=source, target=sink))
```

what about ?

Reply:
You can't += a set, nor a tuple (I just checked)

Reply:
There are a few comment threads about the same lines of code, so I'm going to tag you into the response I just wrote to Paul, and I wrote a similar response further up. 

tl;dr the output of `nx.all_simple_paths(G, source=source, target=sink)` is a list of lists, so turning that into a tuple of lists is not the desired outcome (but I totally get where you're coming from!)